Sample Type Exclusion from Dashboard Insights Panel

cnathe opened the pull request:
#### Rationale
The app dashboard features a Dashboard Insights Panel, which includes bar charts like "Sample Count By Status" and "Sample Count By Sample Type." Clients using Sample Types for inventory purposes have identified a need to exclude certain types from these charts to avoid clutter and enhance data relevance. This set of PRs allow for sample type exclusions from the dashboard graphs in a similar way to how sample types are excluded from Projects (i.e. via the Project Settings page or from the Sample Type Designer).

#### Related Pull Requests
- https://github.com/LabKey/labkey-ui-components/pull/1393
- https://github.com/LabKey/labkey-ui-premium/pull/305
- https://github.com/LabKey/platform/pull/5125
- https://github.com/LabKey/sampleManagement/pull/2381
- https://github.com/LabKey/biologics/pull/2641
- https://github.com/LabKey/inventory/pull/1164

#### Changes
- support for DashboardSampleType as a data type exclusion for projects
- support for DashboardSampleType as a data type exclusion for sample type designer
- ProjectDataTypeSelections panelTitle, panelDescription, and showUncheckedWarning
- DataTypeSelector to load via useFolderDataTypeExclusions hook
- AdminSettingsPage to show Dashboard data type exclusions panel when no projects configured
- CreateProjectPage and ProjectSettings to show Dashboard data type exclusions panel
- Dashboard chart config update to getProjectExclusionFilter to account for an array of exclusion types
- changes sample chart configs to exclude based on both 'SampleType' and 'DashboardSampleType' exclusions
- add sampleTypeDataType to AppAdminContext

cnathe committed:
Support for DashboardSampleType as a data type exclusion for projects
- useFolderDataTypeExclusions hook
- ProjectDataTypeSelections panelTitle, panelDescription, and showUncheckedWarning
- DataTypeSelector to load via useFolderDataTypeExclusions hook

cnathe committed:
AdminSettingsPage to show Dashboard data type exclusions panel when no projects configured

cnathe committed:
3.3.0-fb-dashboardChartDataExclusion.0

cnathe committed:
Merge branch 'develop' into fb_dashboardChartDataExclusion

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json

cnathe committed:
3.4.0-fb-dashboardChartDataExclusion.0

cnathe committed:
Dashboard chart config udpate to getProjectExclusionFilter to account for an array of exclusion types
- changes sample chart configs to exclude based on both 'SampleType' and 'DashboardSampleType' exclusions

cnathe committed:
release notes

cnathe committed:
Remove description from data exclusion sections

cnathe committed:
Fix for issue with containerPath when getting data types for listing

cnathe committed:
Add ProjectDataTypeSelections to ProjectSettings page for Dashboard exclusions

cnathe committed:
3.4.0-fb-dashboardChartDataExclusion.1

cnathe committed:
Merge branch 'develop' into fb_dashboardChartDataExclusion

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
3.5.1-fb-dashboardChartDataExclusion.0

cnathe committed:
Use SampleTypeDataType from adminAppContext to get the expected type filterArray

cnathe committed:
hide project excluded sample types from the dashboard exclusions options

cnathe committed:
add sampleTypeDataType to AppAdminContext

cnathe committed:
CreateProjectPage to show Dashboard data type exclusions panel

cnathe committed:
Create New Project page perm check to useContainerUser() to get homeContainer user object

cnathe committed:
Show app home as a project in ProjectListing to allow for Dashboard sample type exclusions to be set

cnathe committed:
3.5.1-fb-dashboardChartDataExclusion.1

cnathe committed:
update SampleTypeEmptyAlert to include info on excludedDashboardSampleTypes

cnathe committed:
support for DashboardSampleType as a data type exclusion for sample type designer

cnathe committed:
3.5.1-fb-dashboardChartDataExclusion.2

cnathe committed:
Merge branch 'develop' into fb_dashboardChartDataExclusion

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
3.5.4-fb-dashboardChartDataExclusion.0

cnathe committed:
include top level home/app project in DashboardSampleType exclusions

cnathe committed:
3.5.4-fb-dashboardChartDataExclusion.1

XingY commented on packages/components/src/internal/components/project/ProjectListing.tsx:
```diff
@@ -55,7 +54,7 @@ export const ProjectListing: FC<Props> = memo(props => {
                 {projects?.map((project, ind) => {
                     const showInherited = inheritedProjects?.indexOf(project.name) > -1;
-                    let projectTitle = project.path === homeFolderPath ? 'Application' : project.title;
+                    let projectTitle = project.path === HOME_PATH ? HOME_TITLE : project.title;
```
Permissions page used to use "Application" as title for Home project. Is this change intentional?

cnathe replied:
Yes, since we added the home/parent project to the Projects Page, I changed this project title to match as well for consistency.

XingY commented on packages/components/src/internal/components/project/actions.ts:
```diff
@@ -56,6 +57,7 @@ export function getProjectDataTypeDataCount(
         }
 
         Query.executeSql({
```
Shouldn't containerPath get passed in to `getContainerFilterForFolder(containerPath)`?

cnathe replied:
Yes indeed. Good catch. Done.

XingY commented on packages/components/src/internal/components/project/ProjectManagementPage.tsx:
```diff
@@ -138,13 +149,13 @@ export const ProjectManagementPage: FC = memo(() => {
                 <Alert>{error}</Alert>
                 {!loaded && !error && <LoadingSpinner />}
-                {loaded && !error && projects?.length === 0 && (
+                {loaded && !error && projects?.length === 1 && (
```
Would this work for a folder admin who is not a Home project admin?

cnathe replied:
Nope. Good catch. I changed it to a check for at least one project that is not the home project.

XingY commented on packages/components/src/internal/components/project/ProjectDataTypeSelections.tsx:
```diff
@@ -60,35 +81,50 @@ export const ProjectDataTypeSelections: FC<Props> = memo(props => {
+                    <div className="row">
+                        {loaded &&
+                            entityDataTypes?.map(entityDataType => {
+                                const uncheckedDataTypes =
```
maybe add a comment to explain what hidden and unchecked each mean...

cnathe replied:
done

XingY commented on packages/components/src/internal/components/permissions/PermissionAssignments.tsx:
```diff
@@ -420,10 +420,9 @@ export const PermissionAssignments: FC<PermissionAssignmentsProps> = memo(props
-        return 'Application Permissions';
-    }, [selectedProject, moduleContext]);
+        if (!selectedProject) return 'Application Permissions';
+        return (selectedProject.path === HOME_PATH ? HOME_TITLE : selectedProject.name) + ' Permissions';
```
Is the change to not use "Application" as Home project title intentional?

cnathe replied:
Yes, this is for consistency between the Projects Page and the Permissions Page

XingY commented on packages/components/src/internal/components/navigation/FolderMenu.tsx:
```diff
@@ -27,6 +27,10 @@ export const FolderMenu: FC<FolderMenuProps> = memo(props => {
     const { moduleContext, user } = useServerContext();
     const primaryProductId = getPrimaryAppProperties(moduleContext).productId;
 
+    // TODO: the "user" object here is for the current container, so all of the user.isAdmin checks below are incorrect
```
Still TODO?

cnathe replied:
Yes, this was an existing issue and I am not sure exactly how to resolve it so I am going to wait until after this PR to address.

cnathe committed:
AssayProtocolModel.domainContainerId

cnathe committed:
3.5.4-fb-dashboardChartDataExclusion.2

cnathe committed:
jes test fixes and updates

cnathe committed:
3.5.4-fb-dashboardChartDataExclusion.3

cnathe committed:
CR feedback
- getProjectDataTypeDataCount() to pass containerPath to getContainerFilterForFolder()
- add comment regarding hidden entities for related exclusion data type
- fix for Projects Page when user doesn't have perm to home/parent project

cnathe committed:
Merge branch 'develop' into fb_dashboardChartDataExclusion

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
3.6.0-fb-dashboardChartDataExclusion.0

cnathe committed:
jest tests for DashboardSampleType exclusion changes (part 1)

cnathe committed:
Merge branch 'develop' into fb_dashboardChartDataExclusion

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md
#	packages/components/src/internal/components/entities/DataTypeSelector.tsx

cnathe committed:
3.7.1-fb-dashboardChartDataExclusion.0

cnathe committed:
fix for ProjectManagementPage reload on save

cnathe committed:
3.7.1-fb-dashboardChartDataExclusion.1

cnathe committed:
jest tests for DashboardSampleType exclusion changes (part 2)

cnathe committed:
jest tests for DashboardSampleType exclusion changes (part 3)

cnathe committed:
Merge branch 'develop' into fb_dashboardChartDataExclusion

# Conflicts:
#	packages/components/package-lock.json
#	packages/components/package.json
#	packages/components/releaseNotes/components.md

cnathe committed:
3.7.2-fb-dashboardChartDataExclusion.0

cnathe committed:
npm run lint-branch-fix

cnathe committed:
Update release notes with version number and release date

cnathe committed:
3.8.0
